### PR TITLE
Initial drafts of the v0.4 release notes

### PR DESCRIPTION
**Pull request by whitfin**
This is still a WIP, but I wanted to make it available for people to comment on. I'd appreciate if you could focus on anything missing from the features you worked on, so that would mean:

- @nyarly on TLS support
- @colinbankier on Diesel support
- @secretfader on JWT 

And if I've missed anything that you feel deserves mention, please also let me know. I've tried to summarize as much as possible (to the extent I'm familiar), although I haven't had a chance to write up the JWT stuff yet (I only had 1h30m today, I can continue at the weekend).

### CODE REVIEW

File: _posts/2019-06-06-gotham-0.4md [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+
+[Diesel](https://github.com/diesel-rs/diesel) is a very popular ORM
+written in Rust, and there has been some work in progress for a compatible
+middleware for some time. Thanks to the work of `@colinbankier` this
+is now available via the `gotham_middleware_diesel` crate, which offers
+a convenient API for interacting with Diesel from Gotham.
+
+This middleware introduces a `Repo` struct, which is used as a layer
+between Diesel and Gotham to ensure that database interaction lives on
+a separate thread pool to the one connections are being served on. This
```
Comment (colinbankier) [inline _posts/2019-06-06-gotham-0.4md]: I think technically it's not a "separate" thread pool - rather using a tokio feature that allows running blocking operations on the same thread pool without blocking the event loop (and thus other requests). I'm not sure how to word this succinctly, or whether this distinction is important here. 
Maybe just "database interaction does not block handling other requests" or similar?

**Comment by secretfader** [comment]
I'll try to write up a paragraph or two on the JWT middleware that can be included on this post.

**Comment by nyarly** [comment]
I'm :+1:

**Comment by secretfader** [comment]
I'm hoping this is one of those "better late than never" situations. Here's a draft, but feel free to edit.

---

JSON Web Tokens (JWT for short) have become one of the most popular authentication mechanisms for protecting HTTP APIs. That's for good reasons: they are quick and simple to create, reasonably inexpensive to validate, and most of all, tamper resistant. For all of these reasons (and more), we're excited to include middleware to validate JSON Web Tokens in Gotham, version 0.4.

As a user, all you need to do is import the middleware, and pass in a struct with fields required by your application. The middleware implementation does most of the work by validating tokens provided in the HTTP `Authorization` header before passing the data to downstream handlers. If validation fails, an error is returned to the user.

**Comment by whitfin** [comment]
@secretfader perfect, I'll integrate it as soon as I get a chance.

@nyarly @colinbankier after I update it, do you any ideas when we'd like to merge this and have it go public? I can be available this weekend, or later in the day PST throughout the week.

**Comment by nyarly** [comment]
This weekend works for me.

**Comment by colinbankier** [comment]
:+1:

**Comment by secretfader** [comment]
Looks good, @whitfin. :+1:

**Comment by colinbankier** [comment]
Great work, thanks @whitfin :shipit:

**Pull request merged**